hooks/services.py

```diff
@@ -101,15 +101,12 @@ def manage():
         status_set("waiting", msg)
     else:
         manager.manage()
+        cis_misconfigured, cis_message = nrpe_helpers.is_cis_misconfigured()
         if not nrpe_utils.has_consumer():
             status_set("blocked", "Nagios server not configured or related")
         elif nrpe_helpers.has_netlinks_error():
             status_set("blocked", "Netlinks parsing encountered failure; see logs")
-        elif nrpe_helpers.is_cis_misconfigured():
-            status_set(
-                "blocked",
-                "You cannot provide both cis_audit_profile "
-                "and cis_audit_tailoring_file",
-            )
+        elif cis_misconfigured:
+            status_set("blocked", cis_message)
```

Comment:
I found a couple of problems with this validation:

The charm returns to active status after the next update-status event, which means that if quickly loses the blocked status and warning message:

```
$ juju config nrpe cis_audit_profile
foo
$ juju status
...
  nrpe/0*       active    idle            10.37.166.109   icmp,5666/tcp  Ready
```

The validation doesn't prevent the invalid profile from reaching the cron job args:

```
$ juju config nrpe cis_audit_profile=foo
$ juju ssh nrpe/0 cat /etc/cron.d/cis-audit
*/10 * * * * root (/usr/local/lib/nagios/plugins/cron_cis_audit.py -p foo) 2>&1 | logger -t cron_cis_audit
```

Reply:
Regarding the update-status, that seem like a bug to me, so I created #192  to be targeted in a different PR. The code responsible for that is just checking if the nagios service is running and will overwrite all blocks.

Regarding the validation doesn't prevent the profile reaching the con job I think it's not a bad approach. If you pass an invalid profile, the cli responsible for running the cron job will complain, not generating the file and Nagios will alert on this. If we don't let the wrong profile go there and the previous profile was valid, the check will not complain.

Reply:
Thanks, yep that's fair to fix update-status in another PR.  Thanks for opening the bug report. :+1: 

> If you pass an invalid profile, the cli responsible for running the cron job will complain,

Ok that makes sense.  :+1: